tests/unit/filters/test_phase_to_phase.py

```diff
@@ -0,0 +1,128 @@
+# SPDX-FileCopyrightText: 2022 Contributors to the Power Grid Model IO project <dynamic.grid.calculation@alliander.com>
+#
+# SPDX-License-Identifier: MPL-2.0
+import numpy as np
+from power_grid_model.enum import WindingType
+from pytest import approx, mark, raises
+
+from power_grid_model_io.filters.phase_to_phase import (
+    get_clock,
+    get_winding_from,
+    get_winding_to,
+    power_wind_speed,
+    reactive_power,
+    relative_no_load_current,
+)
+
+
+@mark.parametrize(
+    ("i_0", "p_0", "s_nom", "u_nom", "expected"),
+    [
+        (float("nan"), float("nan"), float("nan"), float("nan"), float("nan")),
+        (5.0, 1000.0, 100000.0, 400.0, 0.011547005383792516),
+        (5.0, 2000.0, 100000.0, 400.0, 0.02),
+    ],
+)
+def test_relative_no_load_current(i_0: float, p_0: float, s_nom: float, u_nom: float, expected: float):
+    actual = relative_no_load_current(i_0, p_0, s_nom, u_nom)
+    assert actual == approx(expected) or (np.isnan(actual) and np.isnan(expected))
+
+
+def test_relative_no_load_current__exception():
+    with raises(ValueError, match="can't be more than 100% .* 115.47%"):
+        relative_no_load_current(500.0, 1000.0, 100000.0, 400.0)
+
+
+@mark.parametrize(
+    ("p", "cos_phi", "expected"),
+    [
+        (float("nan"), float("nan"), float("nan")),
+        (1000.0, 0.9, 484.3221048378525),
+    ],
+)
+def test_reactive_power(p: float, cos_phi: float, expected: float):
+    actual = reactive_power(p, cos_phi)
+    assert actual == approx(expected) or (np.isnan(actual) and np.isnan(expected))
+
+
+@mark.parametrize(
+    ("wind_speed", "expected"),
+    [
+        (0.0, 0.0),
+        (1.5, 0.0),
+        (3.0, 0.0),  # cut-in
+        (8.5, 125000.0),
+        (14.0, 1000000.0),  # nominal
+        (19.5, 1000000.0),
+        (25.0, 1000000.0),  # cutting-out
+        (27.5, 500000.0),
+        (30.0, 0.0),  # cut-out
+        (100.0, 0.0),
+    ],
+)
+def test_power_wind_speed(wind_speed, expected):
+    assert power_wind_speed(1e6, wind_speed) == approx(expected)
+
+
+@mark.parametrize(
+    ("code", "winding_type"),
+    [
+        ("Yy1", WindingType.wye),
+        ("Yyn2", WindingType.wye),
+        ("Yd3", WindingType.wye),
+        ("YNy4", WindingType.wye_n),
+        ("YNyn5", WindingType.wye_n),
+        ("YNd6", WindingType.wye_n),
+        ("Dy7", WindingType.delta),
+        ("Dyn8", WindingType.delta),
+        ("Dd9", WindingType.delta),
+    ],
+)
+def test_get_winding_from(code: str, winding_type: WindingType):
+    assert get_winding_from(code) == winding_type
+
+
+def test_get_winding_from__exception():
```

Comment:
double underscore

Reply:
Yes, on purpose. `exception` is a postfix for `test_get_winding_from`.
Some peaple would actually write `test__get_winding_from__exception`, but `test_get_winding_from_exception` is also perfectly valid. I don't have a strong preference...